components/21-atoms/table/table-mobile.functions.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var MobileTable = function (element) { // eslint-disable-line no-unused-vars
+  /**
+   * The table wrapper.
+   */
+  var table = null;
+
+  /**
+   * The table caption.
+   */
+  var caption = null;
+
+  /**
+   * The responsive list version of the table.
+   */
+  var tableList = null;
+
+  /**
+   * Add a wrapper div with class .table-list as next sibling to the table.
+   */
+  var addTableList = function () {
+    var tableListContainer = document.createElement('div');
+    tableListContainer.setAttribute('class', 'table-list');
+    table.parentNode.insertBefore(tableListContainer, tableListContainer.nextSibling);
+
+    tableList = table.nextElementSibling;
+  };
+
+  /**
+   * Adds the individual items to the list <ul></ul> of the table-list wrapper.
+   */
+  var addListItems = function (list) {
+    // Get the headers of the table columns.
+    var headingsNodeList = table.querySelector('table').querySelectorAll('th[scope="col"');
+    // Get all rows of the table.
+    var rowsNodeList = table.querySelector('table').querySelectorAll('tr');
+
+    // Add the rows as a list item.
+    for (var i = 0; i < rowsNodeList.length; i++) {
+      var row = rowsNodeList[i];
+
+      // Get all the columns of the table based on the current row.
+      var colsNodeList = row.querySelectorAll('td');
```

Comment:
You assume that a `<tr>` always consists of only `<td>`'s, while it's perfectly possible to have `<th>` tags in there as well. 

It might be safer to loop over all children, then check for `<th>` vs. `<td>` tags.

```suggestion
      var colsNodeList = row.childNodes;
```

Reply:
Th's are not used in this loop... So this won't work.